modules/core/src/localstorageexpiry.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { IdleExpiry } from './idleexpiry';
+import { LocalStorage } from './localstorage';
+
+/*
+ * Represents a localStorage store of expiry values.
+ * @extends IdleExpiry
+ */
+@Injectable()
+export class LocalStorageExpiry extends IdleExpiry {
+
+  constructor(private localStorage: LocalStorage) {
+    super();
+  }
+
+  /*
+   * Gets or sets the last expiry date in localStorage.
+   * If localStorage doesn't work correctly (i.e. Safari in private mode), we store the expiry value in memory.
+   * @param value - The expiry value to set; omit to only return the value.
+   * @return The current expiry value.
+   */
+  last(value?: Date): Date {
+    if (value !== void 0) {
+      this.setExpiry(value);
+    }
+    return this.getExpiry();
+  }
+
+  private getExpiry(): Date {
+    let expiry: string = this.localStorage.getItem('expiry');
```

Comment:
One thing people have been asking for from ng-idle was the ability to change the key so that if they had multiple Angular apps in one domain (or, as a new challenge now that we can support multiple `Idle` instances per app) they don't conflict.

We could allow for this by taking in a new constructor parameter `key: string` (defaulting to the current `expiry`) and use that when getting or setting `localStorage`. The developer could then opt-in to this feature by   initializing and providing a new instance rather than the current default `LocalStorageExpiry`.